File: Graduation_Project_FE/shared/api/services/invoiceService.ts
import { axiosClient } from '../axiosClient';
import { ENDPOINTS } from '../endpoints';

export interface InvoiceRequest {
  invoiceName: string;
  amount: number;
  dueDate: string;
  reminderOption: string;
  reminderTime?: string;
  isPaid: boolean;
}

export interface InvoiceResponse {
  id: number;
  invoiceName: string;
  amount: number;
  dueDate: string;
  reminderOption: string;
  reminderTime?: string;
  isPaid: boolean;
  createdAt: string;
  updatedAt: string;
}

export const invoiceService = {
  createInvoice: async (data: InvoiceRequest) => {
    const response = await axiosClient.post(ENDPOINTS.INVOICE.BASE, data);
    return response;
  },

  getInvoices: async () => {
    const response = await axiosClient.get(ENDPOINTS.INVOICE.BASE);
    return response;
  },

  getInvoiceById: async (id: number) => {
    const response = await axiosClient.get(ENDPOINTS.INVOICE.DETAIL(id));
    return response;
  },

  updateInvoice: async (id: number, data: InvoiceRequest) => {
    const response = await axiosClient.put(ENDPOINTS.INVOICE.DETAIL(id), data);
    return response;
  },

  updateInvoiceStatus: async (id: number, isPaid: boolean) => {
    const response = await axiosClient.patch(ENDPOINTS.INVOICE.STATUS(id), null, {
      params: { isPaid }
    });
    return response;
  },

  deleteInvoice: async (id: number) => {
    const response = await axiosClient.delete(ENDPOINTS.INVOICE.DETAIL(id));
    return response;
  }
};

File: Graduation_Project_BE/api-app-customer/src/main/java/com/project/app/invoice/service/InvoiceService.java
package com.project.app.invoice.service;

import com.project.app.invoice.dto.request.InvoiceRequest;
import com.project.app.invoice.dto.response.InvoiceResponse;
import com.project.app.user.entity.User;

import java.util.List;

public interface InvoiceService {
    InvoiceResponse createInvoice(User user, InvoiceRequest request);
    List<InvoiceResponse> getInvoices(User user);
    InvoiceResponse getInvoiceById(Long id, User user);
    InvoiceResponse updateInvoice(Long id, User user, InvoiceRequest request);
    void deleteInvoice(Long id, User user);
    InvoiceResponse updateInvoiceStatus(Long id, User user, boolean isPaid);
    InvoiceResponse payInvoiceWithCash(Long id, User user);
}
